fixieai/agents/code_shot.py

```diff
@@ -211,32 +222,45 @@ def _serve_func(
                 f"Func[{func_name}] returned unexpected output of type {type(output)}."
             )
 
-    def _verify_token(self, token: str) -> bool:
+    def _verify_token(self, token: str) -> Optional[VerifiedTokenClaims]:
         try:
             public_key = self._jwks_client.get_signing_key_from_jwt(token)
-            _ = jwt.decode(
+            claims = jwt.decode(
                 token,
                 public_key.key,
                 algorithms=["EdDSA"],
-                audience=constants.FIXIE_AGENT_API_AUDIENCE,
+                audience=constants.FIXIE_AGENT_API_AUDIENCES,
             )
-            return True
+
+            if _AGENT_ID_JWT_CLAIM not in claims or not isinstance(
```

Comment:
nit: I would put this logic into a `VerifiedTokenClaims.from_jwt(token, public_key)` that raises `jwt.DecodeError` if bad format. And probably move that dataclass to further down in this file (since this is in SDK and the order will become the order in the docs).

Reply:
I ended up consolidating all token validation into `_VerifiedTokenClaims` and passing the client in to skirt around the awkwardness of raising `jwt.DecodeError`